src/OpenLoco/src/Map/MapGenerator.cpp

```diff
@@ -392,6 +400,112 @@ namespace OpenLoco::World::MapGenerator
         }
     };
 
+    class HeightMapTerrainGenerator
+    {
+    public:
+        void generate(HeightMapRange heightMap)
+        {
+            generateFromHeightmapPng(heightMap);
+        }
+
+    private:
+        void getPixel(const std::vector<png_byte>& imageData, int x, int y, png_byte& red, png_byte& green, png_byte& blue, png_byte& alpha, int width)
+        {
+            int index = (y * width + x) * 4;
+            red = imageData[index];
+            green = imageData[index + 1];
+            blue = imageData[index + 2];
+            alpha = imageData[index + 3];
+        }
+
+        bool openUiPngBrowser()
+        {
+            auto basePath = Platform::getUserDirectory();
+
+            // open ui prompt window
+            // doesn't work...
+            bool confirm = Ui::Windows::PromptBrowse::open(OpenLoco::Ui::Windows::PromptBrowse::browse_type::load, (char*)basePath.string().c_str(), "*.png", StringIds::title_prompt_load_game);
+
+            return confirm;
+        }
+
+        // note: png image must be 384x384 or otherwise this method will probably crash
+        int generateFromHeightmapPng(HeightMapRange heightMap)
+        {
+            auto basePath = Platform::getUserDirectory();
+            auto defaultHeightmap = basePath / "heightmap.png";
+            auto filename = defaultHeightmap.string();
+
+            // todo:
+            /*if (openUiPngBrowser())
+            {
+                filename = std::string(_savePath);
+            }*/
+
+            std::ifstream inFile(filename, std::ios::binary);
+
+            png_structp png = png_create_read_struct(PNG_LIBPNG_VER_STRING, NULL, NULL, NULL);
```

Comment:
this png file loading should go in a separate class dedicated to handling images. same for the saving of  images in screenshot code - move it elsewhere into a 'png' class of some kind